grpc/rust/BUILD

```diff
@@ -85,3 +86,44 @@ checkstyle_test(
     license_type = "agpl-header",
     size = "small",
 )
+
+cargo_build_script(
+    name = "build_script",
+    srcs = ["build.rs"],
+    data = [
+        "//cluster:service-proto",
+        "//cluster:server-proto",
+        "//cluster:user-proto",
+        "//cluster:database-proto",
+        "//common:answer-proto",
+        "//common:concept-proto",
+        "//common:logic-proto",
+        "//common:options-proto",
+        "//common:query-proto",
+        "//common:session-proto",
+        "//common:transaction-proto",
+        "//core:database-proto",
+        "//core:service-proto",
+        "@com_google_protobuf//:protoc",
+    ],
+    build_script_env = {
+        "PROTOC": "$(execpath @com_google_protobuf//:protoc)"
+    },
+    deps = [
+        "@vaticle_dependencies//library/crates:tonic_build",
+    ],
+)
+
+rust_library(
+    name = "lib",
```

Comment:
Let's rename this to `typedb-protocol`, for consistency with Java, Python and Node.js.

This is in line with our other library repos, where the "entry point" - i.e the principal export of the repo - is named the same as the repo.

Reply:
Done.